### DISCUSSION TITLE
DOMPDF RENDER

### DISCUSSION THREAD

**Discussion by NirmalBalajiTU** [dompdf/dompdf#3448] [General]
Rendering takes huge time.

**Comment by bsweeney** [comment]
We would need more details about the document you're attempting to render before anybody could provide any kind of useful guidance.

**Reply by NirmalBalajiTU** [reply]
When the file size increases, rendering time significantly extends. For instance, if a 200kb file renders in 4 seconds, an 800kb file ideally renders in 16 seconds. However, it's taking more than that (36 seconds), with the design remaining unchanged. Despite extensive searching, I haven't found a solution.

1.Removing all styles didn't improve rendering time.
2.Attempting to split the HTML content didn't resolve the issue, as ultimately, merging all files is necessary.
Honestly, I've tried other libraries that render faster, but they don't meet all my requirements. Any ideas on optimizing the current library's performance would be greatly appreciated.

**Reply by NirmalBalajiTU** [reply]
And it is a static table structure with width, border and text align styles not much complicated.

**Reply by bsweeney** [reply]
The biggest issue there is just the usage of tables. Dompdf has some serious performance issues tracking table layout. It's something I'd like to address but I'm not sure when that would be done. I haven't done any performance testing, but you might see better performance if you can break the table up into multiple tables.

**Reply by NirmalBalajiTU** [reply]
After spending a lot of time in dompdf, I have made some mistakes:

If your PDF doesn't have images, don't enable remote, as it will take some time. You can reduce that time by not enabling it.
If you're using images, make sure the size is compressed as much as possible.
This will definitely run much faster.
Split the PDF. For example, if you have a 300-page PDF, split it into smaller files, like every 30 pages, and convert them to HTML.
After splitting it into HTML, convert each part to PDF using dompdf.
Finally, use a PDF merger to combine the files. It will merge them much faster.
Before, it took 30 minutes; after these changes, it takes only 1.5 minutes."
